packages/canary-web-components/src/components/ic-card-horizontal/ic-card-horizontal.tsx

```diff
@@ -0,0 +1,249 @@
+
+    if (this.parentEl.tagName === "A") {
+      this.clickable = true;
+      this.parentIsAnchorTag = true;
+      this.parentEl.classList.add("ic-card-wrapper-link");
+      this.parentEl.addEventListener("focus", this.parentFocussed);
+      this.parentEl.addEventListener("blur", this.parentBlurred);
+    }
+    removeDisabledFalse(this.disabled, this.el);
+  }
+
+  componentDidLoad(): void {
+    !isSlotUsed(this.el, "heading") &&
+      onComponentRequiredPropUndefined(
+        [{ prop: this.heading, propName: "heading" }],
+        "Card"
+      );
+    this.updateTheme();
+  }
+
+  @Listen("click", { capture: true })
+  handleHostClick(event: Event): void {
+    if (this.disabled) {
+      event.stopImmediatePropagation();
+    }
+  }
+
+  @Listen("themeChange", { target: "document" })
+  themeChangeHandler(ev: CustomEvent): void {
+    const theme: IcTheme = ev.detail;
+    this.updateTheme(theme.mode);
+  }
+
+  /**
+   * Sets focus on the card.
+   */
+  @Method()
+  async setFocus(): Promise<void> {
+    if (this.el.shadowRoot.querySelector("a")) {
+      this.el.shadowRoot.querySelector("a").focus();
+    } else if (this.el.shadowRoot.querySelector("button")) {
+      this.el.shadowRoot.querySelector("button").focus();
+    }
+  }
+
+  private parentFocussed = (): void => {
+    this.isFocussed = true;
+  };
+
+  private parentBlurred = (): void => {
+    this.isFocussed = false;
+  };
+
+  private updateTheme(newTheme: IcThemeForeground = null): void {
+    const foregroundColor = getThemeFromContext(this.el, newTheme || null);
+
+    if (foregroundColor !== IcThemeForegroundEnum.Default) {
+      this.appearance = foregroundColor;
+    }
+  }
+
+  render() {
+    const {
+      clickable,
+      disabled,
+      heading,
+      message,
+      href,
+      hreflang,
+      referrerpolicy,
+      rel,
+      target,
+      parentIsAnchorTag,
+      isFocussed,
+      size,
+    } = this;
+
+    const Component = parentIsAnchorTag
+      ? "div"
+      : clickable
+      ? this.href === undefined
+        ? "button"
+        : "a"
+      : "div";
+
+    const attrs = Component == "a" && {
+      href: href,
+      hrefLang: hreflang,
+      referrerPolicy: referrerpolicy,
+      rel: rel,
+      target: target,
+    };
+
+    return (
+      <Component
+        class={{
+          ["card"]: true,
+          ["clickable"]: clickable && !disabled,
+          ["disabled"]: disabled,
+          ["focussed"]: isFocussed,
+          ["dark"]: this.appearance === IcThemeForegroundEnum.Dark,
+          [`${size}`]: true,
+          ["with-icon"]: isSlotUsed(this.el, "icon"),
+          ["with-image"]: isSlotUsed(this.el, "image"),
+        }}
+        tabindex={clickable && !parentIsAnchorTag ? 0 : null}
+        aria-disabled={disabled ? "true" : null}
+        disabled={disabled ? true : null}
+        {...attrs}
+      >
+        {isSlotUsed(this.el, "image") && (
+          <div class={{ ["image"]: true }}>
```

Comment:
Tested in the React Storybook Playground and the card doesn't resize if the image slot is removed/added after first render. Other props seem fine :)